balancer/leastrequest/leastrequest.go

```diff
@@ -0,0 +1,182 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package leastrequest
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/balancer/base"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/internal/grpcrand"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+// Global to stub out in tests.
+var grpcranduint32 = grpcrand.Uint32
+
+// Name is the name of the least request balancer.
+const Name = "least_request_experimental"
+
+var logger = grpclog.Component("least request")
```

Comment:
Nit: `least-request`.  One day components will be configurable, with env vars, and spaces will be unfriendly.

Reply:
Done.